packages/relayer/modules/attested-to-cosmos/src/tx_builder.rs

```diff
@@ -0,0 +1,249 @@
+                packet,
+                cyphon,
+                &src_client_id,
+                &dst_client_id,
+                seqs,
+            );
+        }
+
+        let query_height = *heights.iter().max().unwrap();
+        let request = aggregator_proto::GetAttestationsRequest {
+            packets: ics26_send_packets,
+            height: query_height, // latest height
+        };
+
+        tracing::info!(
+            "Requesting state attestation from aggregator for {} packets",
+            request.packets.len()
+        );
+        // - We need update client with timestamp at height
+        // - We need MsgRecvPacket where proof is (key, value, height, signature)
+        let response = aggregator_client
+            .get_attestations(request)
+            .await?
+            .into_inner();
+
+        let (state, packets) = (
+            response
+                .state_attestation
+                .ok_or(anyhow::anyhow!("No state received"))?,
+            response
+                .packet_attestation
+                .ok_or(anyhow::anyhow!("No packets received"))?,
+        );
+        tracing::info!(
+            "Received state attestation: {} signatures, height {}, state: {}",
+            packets.signatures.len(),
+            packets.height,
+            hex::encode(&packets.attested_data)
+        );
+
+        let header = Header::new(
+            state.height,
+            state.timestamp.unwrap(),
+            state.attested_data,
+            state.signatures,
+            state.public_keys,
+        );
+        let header_bz = serde_json::to_vec(&header)
+            .map_err(|_| anyhow::anyhow!("header could not be serialized"))?;
+
+        let update_msg = MsgUpdateClient {
+            client_id: dst_client_id.clone(),
+            client_message: Some(Any::from_msg(&ClientMessage { data: header_bz })?),
+            signer: self.signer_address.clone(),
+        };
+
+        let now_since_unix = std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH)?;
+
+        let mut timeout_msgs = cosmos::target_events_to_timeout_msgs(
+            target_events,
+            &src_client_id,
+            &dst_client_id,
+            &dst_packet_seqs,
+            &self.signer_address,
+            now_since_unix.as_secs(),
+        );
+
+        let (mut recv_msgs, mut ack_msgs) = cosmos::src_events_to_recv_and_ack_msgs(
+            src_events,
+            &src_client_id,
+            &dst_client_id,
+            &src_packet_seqs,
+            &dst_packet_seqs,
+            &self.signer_address,
+            now_since_unix.as_secs(),
+        );
+
+        tracing::debug!("Timeout messages: #{}", timeout_msgs.len());
+        tracing::debug!("Recv messages: #{}", recv_msgs.len());
+        tracing::debug!("Ack messages: #{}", ack_msgs.len());
+
+        attestor::inject_proofs(&mut recv_msgs, &packets.attested_data, packets.height);
+
+        // We don't want to use mock roofs for RecvMsg
```

Comment:
Would prefer to mark this with a TODO and the issue it is meant to be implement in